src/Sylius/Bundle/RbacBundle/Form/EventSubscriber/AddParentFormSubscriber.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\RbacBundle\Form\EventSubscriber;
+
+use Sylius\Bundle\RbacBundle\Form\Exception\UnexpectedTypeException;
+use Sylius\Component\Rbac\Model\PermissionInterface;
+use Sylius\Component\Rbac\Model\RoleInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class AddParentFormSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @param string $type
+     */
+    public function __construct($type)
+    {
+        $this->type  = $type;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return array(
+            FormEvents::PRE_SET_DATA => 'preSetData',
+        );
+    }
+
+    /**
+     * @param FormEvent $event
+     */
+    public function preSetData(FormEvent $event)
+    {
+        $resource = $event->getData();
+
+        if (null === $resource) {
+            return;
+        }
+
+        if (!$resource instanceof RoleInterface && !$resource instanceof PermissionInterface) {
+            throw new UnexpectedTypeException($resource, RoleInterface::class, PermissionInterface::class);
+        }
+
+        if (null === $resource->getId() XOR null === $resource->getParent()) {
```

Comment:
This condition changes the previous behavior:

ConditionBefore (remove parent when true):
`null !== $permission->getId() && null === $permission->getParent()`
ConditionNow (add parent when false):
`null === $resource->getId() XOR null === $resource->getParent()`

| ID | Parent | ResultBefore | ResultNow |
|----|---|---|---|
| null | null | add parent | add parent
| null | not null | add parent | no parent
| not null | null | no parent | no parent
| not null | not null | add parent | add parent


The proper condition should be:
`null !== $resource->getId() && null === $resource->getParent()`